cmake/TestSetup.cmake

```diff
@@ -126,7 +133,12 @@ function(setup_test_mpi nproc dir arg extra_labels)
       --test_dir ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}
       --output_dir "${PROJECT_BINARY_DIR}/tst/regression/outputs/${dir}_mpi"
       --kokkos_args=${PARTHENON_KOKKOS_TEST_ARGS})
-    set_tests_properties(regression_mpi_test:${dir} PROPERTIES LABELS "${labels}" RUN_SERIAL ON )
+    set_tests_properties(
+      regression_mpi_test:${dir}
+      PROPERTIES
+        LABELS "${labels}"
```

Comment:
What happened to the RUN_SERIAL option?

Reply:
I think that logic was moved to the setup_test function above.

Reply:
Well, not quite - the intention with setting PROCESSORS was to be better than RUN_SERIAL, since it lets you still get parallel test execution, but on different processors.

However, my one concern with this was the interaction with GPU execution.

Reply:
I'm thinking I should keep `RUN_SERIAL` if `Kokkos_ENABLE_CUDA` since cmake can't really control "GPU affinity". Thoughts?

This change makes the tests run a lot faster on CPU builds, so I'd like to keep it there.

Reply:
That's fine by me.